docs/DirectoryManager.md

```diff
@@ -0,0 +1,127 @@
+
+## DirectoryManager Implementation in Apps Script
+
+Therefore a lot of thought has gone into the design of this module. The design aims to provide maximum support for Windows VBA API usage. Some legacy and deprecated API and features are not being implemented.
+
+### `CurDir()`
+
+Official Documentation - https://docs.microsoft.com/en-us/office/vba/language/reference/user-interface-help/curdir-function
+
+**VBA API Usage:**
+
+`CurDir [ (drive) ]`
+
+Design:
+
+1. `CurDir` is set to `ActiveWorkbook.path` in the beginning of each macro call. This behaviour differs from Windows but this has been decided as this is the most expected usage. 
+1. Changes to CurDir does not persist across calls
+1. If a Drive is passed to `CurDir` then, if the drive is same as current working directory, then just return it. If the drive passed is different, then just return root directory. For example `CurDir("E")`, return `E:\`
+
+
+### `ChDir()`
+
+Official Documentation - https://docs.microsoft.com/en-us/office/vba/language/reference/user-interface-help/chdir-statement
+
+**VBA API Usage:**
+
+`ChDir path`
+
+Design:
+
+1. `ChDir()` changes will not persist across calls because `CurDir()` does not persist across calls.
+1. Will not support legacy Macintosh path type: `ChDir "MacDrive:Tmp"`
+1. Will support windows style call type for both mac and windows file paths
+1. `ChDir ".."` will move up one directory on both mac and windows.
+
+### `ChDrive()`
+
+Official Documentation - https://docs.microsoft.com/en-us/office/vba/language/reference/user-interface-help/chdrive-statement
+
+**VBA API Usage:**
+
+`ChDrive drive`
+
+Design:
+
+1. `ChDrive()` changes will not persist across calls because `CurDir()` does not persist across calls.
+1. If the drive is passed to `ChDrive` is same as current working directory, then don't do anything. If the drive passed is different, then just set current working directory to root directory. For example `ChDrive("E")`, set current working directory to `E:\`
+
+### `Dir()`
+
+Official Documentation - https://docs.microsoft.com/en-us/office/vba/language/reference/user-interface-help/dir-function
+
+**VBA API Usage:**
+
+Dir [ (pathname, [ attributes ] ) ]
+
+| Part | Description | Implemented | Comment |
+|-|-|-|-|
+| pathname | Optional. String expression that specifies a file name; may include directory or folder, and drive. A zero-length string ("") is returned if pathname is not found. | Yes | Both relative and absolute path work |
+| attributes | Optional. Constant or numeric expression, whose sum specifies file attributes. If omitted, returns files that match pathname but have no attributes. | No | Cannot implement as it is windows file system specific |
+
+Design:
+
+1. Both relative and absolute paths work.
+1. Dir can also take in wild cards. "?" for single character and "*" for multiple characters.
+1. Dir() without a parameter works like a generator, returns the next available match. 
```

Comment:
What is next available match? Ambiguous.

Reply:
I added an example, should be clear now